crates/db_schema/src/impls/instance.rs

```diff
@@ -46,12 +46,32 @@ impl Instance {
     let conn = &mut get_conn(pool).await?;
     Self::read_or_create_with_conn(conn, domain).await
   }
+
   pub async fn delete(pool: &DbPool, instance_id: InstanceId) -> Result<usize, Error> {
     let conn = &mut get_conn(pool).await?;
     diesel::delete(instance::table.find(instance_id))
       .execute(conn)
       .await
   }
+
+  pub async fn read_all(pool: &DbPool) -> Result<Vec<Instance>, Error> {
+    let conn = &mut get_conn(pool).await?;
+    instance::table
+      .select(instance::all_columns)
+      .get_results(conn)
+      .await
+  }
+
+  pub async fn dead_instances(pool: &DbPool) -> Result<Vec<String>, Error> {
+    let conn = &mut get_conn(pool).await?;
+    instance::table
+      .select(instance::domain)
+      // TODO: should use instance::published if updated is null
+      //.filter(instance::updated.lt(now - 3.days()))
```

Comment:
Not sure how to write this query.

Reply:
COALESCE? coalesce(instance::updated, instance::published).lt(now - 3.days())

https://diesel.rs/guides/extending-diesel.html

use diesel::sql_types::{Nullable, Text};
sql_function! { fn coalesce(x: Nullable<Text>, y: Text) -> Text; }

Reply:
That works, thanks! Although I dont see how to make it generic, and have to write it specifically for timestamp type.